src/metaopt/worker/experiment.py

```diff
@@ -0,0 +1,354 @@
+# -*- coding: utf-8 -*-
+"""
+:mod:`metaopt.worker.experiment` -- Description of an optimization attempt
+==========================================================================
+
+.. module:: trial
+   :platform: Unix
+   :synopsis: Manage history of trials corresponding to a black box process
+
+"""
+
+import copy
+import datetime
+import getpass
+import logging
+import math
+import random
+
+import six
+
+from metaopt.io.database import Database
+from metaopt.worker.trial import Trial
+
+log = logging.getLogger(__name__)
+
+
+class Experiment(object):
+    """Represents an entry in database/experiments collection.
+
+    Attributes
+    ----------
+    name : str
+       Unique identifier for this experiment per `user`.
+    refers : dict or list of `Experiment` objects, after initialization is done.
+       A dictionary pointing to a past `Experiment` name, ``refers[name]``, whose
+       trials we want to add in the history of completed trials we want to re-use.
+    metadata : dict
+       Contains managerial information about this `Experiment`.
+    pool_size : int
+       How many workers can participate asynchronously in this `Experiment`.
+    max_trials : int
+       How many trials must be evaluated, before considering this `Experiment` done.
+
+       This attribute can be updated if the rest of the experiment configuration
+       is the same. In that case, if trying to set to an already set experiment,
+       it will overwrite the previous one.
+    status : str
+       A keyword among {*'new'*, *'running'*, *'done'*, *'broken'*} indicating
```

Comment:
So `'pending': Denotes an experiment with valid configuration which is currently being handled by **MetaOpt**` ?

Reply:
```pending``` denotes an experiment with valid configuration which is not done or broken. It might have trials running or not. In some sense, ```pending == new + running```